### ISSUE TITLE
Request deadline expires while queued; expiry returns 1 token as finish_reason=length, indistinguishable from max_tokens

### ISSUE THREAD

**Issue by AzeezIsh** [Avarok-Cybersecurity/atlas#482]
A request whose `--request-timeout` deadline expires **while it is still queued**
is prefilled anyway, killed on its first decode step, and returned to the client
as a normal `200` with `finish_reason: "length"` and a ~1-token body. There is no
field, status code, or response marker distinguishing it from a legitimate
`max_tokens` stop.

That makes it silent data corruption for any eval harness: the truncated output
scores as a *wrong answer* rather than an error, so nothing retries and nothing
flags it. It cost a full 5-hour BFCL run here before I noticed.

## What happens

`timeout_at` is stamped at **arrival** (`api/completions_exec.rs:121`,
`api/chat/sampling_setup.rs:310`), but `request_start` is stamped at **prefill**
(`scheduler/prefill_a_step.rs:120`). Under queueing those diverge: the deadline
is consumed by queue wait, so by the time the sequence reaches
`decode_logits_step.rs:721` it is already past due and dies having produced one
token.

Measured over one run (1294 requests, single server process):

| | requests | median output tokens | finish_reason |
|---|---:|---:|---|
| timed out | 1126 | **1** | `length` |
| clean | 168 | 96 | `stop` / `tool_calls` |

The first timeout fires 303 s into the run and essentially every request after it
times out — once the backlog exceeds the timeout, the server enters a stable
state where it spends its entire time prefilling requests that are already dead.
It never recovers on its own, and throughput looks fine the whole time.

## Reproduce

Any client concurrency above what `--max-batch-size` can absorb will do. Mine was
`--max-batch-size 1` against BFCL (`bfcl-eval 2025.8.6.2`). Worth knowing if you
benchmark with it: `--num-threads` is honoured on the hosted-API path
(`_llm_response_generation.py:248`, `max_workers=args.num_threads`) but **not** on
the local/OSS path — `base_oss_handler.py:222` hardcodes
`ThreadPoolExecutor(max_workers=100)`. So `--num-threads 1` against a local server
still fires 100 concurrent requests, which is how I ended up here believing the
run was serial. Dropping that to one worker changed timeouts from 1126/1294 to
**0** with no server-side change.

A direct version: start with `--request-timeout 1` and send one request. The
deadline expires during prefill, and the response comes back `finish_reason:
"length"` with a 1-token body.

## The misleading part

The warning reports the wrong clock:

```rust
tracing::warn!("Request timeout after {:?}", a.request_start.elapsed());
```

`request_start` is prefill, so a 300 s budget blown in the queue prints as
`Request timeout after 8.4s`. That reads as "generation timed out after 8 s",
which sends you looking at decode speed. The real signal — that the request
produced 1 token and was already over budget before decode began — is not in the
message at all.

## Suggestions

1. **Make expiry distinguishable.** At minimum set `guard_stop =
   Some("request_timeout")` (the existing mechanism, as `fuzzy_repetition` does)
   — though note `guard_stop` is currently dropped on the blocking
   `/v1/completions` path (`api/completions.rs:424`), so it would need to be
   surfaced there to help this case. A distinct `finish_reason` would be
   stronger; an HTTP error for "deadline expired before any output" stronger
   still, since that result is never useful to a caller.
2. **Don't prefill a request that is already past its deadline.** Checking at
   promote time would let a backlogged server shed load and recover instead of
   burning every prefill slot on dead requests.
3. **Consider whether queue wait should count against the deadline at all**, or
   whether it wants to be a separate queue-wait budget — a request that waits
   299 s and then generates perfectly still gets killed one token in.

PR #470 (branch `windows-hip-first-run`) carries the diagnostic half of this: the
warning now reports output tokens, time since prefill, and how far past the
deadline it is, and sets `guard_stop`. The behavioural items above are
deliberately not in it.

Found on a Framework Desktop (Ryzen AI MAX+ 395 / gfx1151, Windows 11) serving
`nvidia/Qwen3.6-27B-NVFP4`, but nothing here looks Windows- or model-specific.
